File: Veiculo.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class VeiculoTest {
    
    @Test
    void testAcelerar() {
        Veiculo veiculo = new Veiculo("Carro", "Ford", 2023, 0);
        veiculo.acelerar(20);
        assertEquals(20, veiculo.getVelocidade());
    }

    @Test
    void testFrear() {
        Veiculo veiculo = new Veiculo("Carro", "Ford", 2023, 50);
        veiculo.frear(20);
        assertEquals(30, veiculo.getVelocidade());
    }

    @Test
    void testParar() {
        Veiculo veiculo = new Veiculo("Carro", "Ford", 2023, 50);
        veiculo.parar();
        assertEquals(0, veiculo.getVelocidade());
    }
}


public class Veiculo {
    private String tipo;
    private String marca;
    private int ano;
    private int velocidade;

    public Veiculo(String tipo, String marca, int ano, int velocidade) {
        this.tipo = tipo;
        this.marca = marca;
        this.ano = ano;
        this.velocidade = velocidade;
    }

    public void acelerar(int incremento) {
        this.velocidade += incremento;
    }

    public void frear(int decremento) {
        this.velocidade -= decremento;
        if (this.velocidade < 0) {
            this.velocidade = 0;
        }
    }

    public void parar() {
        this.velocidade = 0;
    }

    public int getVelocidade() {
        return velocidade;
    }
}

File: Veiculo.py
import unittest

class TestVeiculo(unittest.TestCase):
    def setUp(self):
        self.veiculo = Veiculo("Carro", "Ford", 2023, 0)

    def test_acelerar(self):
        self.veiculo.acelerar(20)
        self.assertEqual(self.veiculo.velocidade, 20)

    def test_frear(self):
        self.veiculo.acelerar(50)
        self.veiculo.frear(20)
        self.assertEqual(self.veiculo.velocidade, 30)

    def test_parar(self):
        self.veiculo.acelerar(50)
        self.veiculo.parar()
        self.assertEqual(self.veiculo.velocidade, 0)

if _name_ == "_main_":
    unittest.main()


class Veiculo:
    def _init_(self, tipo, marca, ano, velocidade):
        self.tipo = tipo
        self.marca = marca
        self.ano = ano
        self.velocidade = velocidade

    def acelerar(self, incremento):
        self.velocidade += incremento

    def frear(self, decremento):
        self.velocidade -= decremento
        if self.velocidade < 0:
            self.velocidade = 0

    def parar(self):
        self.velocidade = 0